cli/azd/pkg/config/config.go

```diff
@@ -71,22 +70,26 @@ func (c *config) Raw() map[string]any {
 }
 
 // SetSecret stores the secrets at the specified path within a local user vault
-func (c *config) SetSecret(path string, value string) error {
+func (c *config) SetSecret(path string, value string) (string, error) {
 	if c.vaultId == "" {
 		c.vault = NewConfig(nil)
 		c.vaultId = uuid.New().String()
 		if err := c.Set("vault", c.vaultId); err != nil {
-			return fmt.Errorf("failed setting vault id: %w", err)
+			return "", fmt.Errorf("failed setting vault id: %w", err)
 		}
 	}
 
-	pathId := uuid.New().String()
-	vaultRef := fmt.Sprintf("vault://%s/%s", c.vaultId, pathId)
-	if err := c.vault.Set(pathId, base64.StdEncoding.EncodeToString([]byte(value))); err != nil {
-		return fmt.Errorf("failed setting secret value: %w", err)
+	secretId := uuid.New().String()
+	vaultRef := fmt.Sprintf("vault://%s/%s", c.vaultId, secretId)
+	if err := c.vault.Set(secretId, base64.StdEncoding.EncodeToString([]byte(value))); err != nil {
+		return "", fmt.Errorf("failed setting secret value: %w", err)
+	}
+
+	if err := c.Set(path, vaultRef); err != nil {
+		return "", fmt.Errorf("failed setting secret reference: %w", err)
 	}
 
-	return c.Set(path, vaultRef)
+	return vaultRef, nil
```

Comment:
What can folks do with a `vaut://id/id` ref?

Reply:
I am returning the values from the `config.SetSecret()` call so that the environment can store that vault reference in the raw `.env` file.

Reply:
Can we just have .env and config with normal keys and let the Config to handle merging the 2?  
There's no benefit to customers by looking at the vault references... All they need is the vault id